scanner/src/main/java/ca/on/oicr/gsi/runscanner/scanner/processor/RevioPacBioProcessor.java

```diff
@@ -247,39 +207,40 @@ public NotificationDto process(File runDirectory, TimeZone tz) throws IOExceptio
               .filter(file -> PB_REPORT_LOG.test(String.valueOf(file.getFileName())))
               .toList();
 
+      // Grab completion time from a file
       if (logFiles.size() == smrtCellCount) {
-        // Grab completion time from log file
-        Optional<File> mostRecentlyCompleted =
-            getSubDirectory(runDirectory, "statistics")
+        Optional<Instant> latestCompletionTime =
+            streamSmrtCellSubdirectories(runDirectory, "statistics")
                 .flatMap(
                     statisticsDirectory ->
                         Stream.of(
-                            Objects.requireNonNull(
-                                statisticsDirectory.listFiles(
-                                    file -> PB_REPORT_LOG.test(file.getName())))))
-                .max(Comparator.comparing(RevioPacBioProcessor::getLogCompletionTime));
-        // Set completion time based on log file timestamp
-        dto.setCompletionDate(getLogCompletionTime(mostRecentlyCompleted.orElse(null)));
-
+                            statisticsDirectory.listFiles(
+                                file -> PB_REPORT_LOG.test(file.getName()))))
+                .map(RevioPacBioProcessor::getLogCompletionTime)
+                .max(Comparator.naturalOrder());
+        // Set completion time based on pbereports.log file
+        dto.setCompletionDate(latestCompletionTime.get());
+      } else {
         // We don't have the pbereport.log, we'll have to fallback on using transfer_done file
         // creation time instead
         if (dto.getCompletionDate() == null) {
-          Optional<File> mostRecentDoneFile =
-              getSubDirectory(runDirectory, "metadata")
+          Optional<Instant> mostRecentDoneFile =
+              streamSmrtCellSubdirectories(runDirectory, "metadata")
                   .flatMap(
                       metadataDirectory ->
                           Stream.of(
-                              Objects.requireNonNull(
-                                  metadataDirectory.listFiles(
-                                      file -> TRANSFER_DONE.test(file.getName())))))
-                  .max(Comparator.comparing(RevioPacBioProcessor::getFileCreationTime));
+                              metadataDirectory.listFiles(
+                                  file -> TRANSFER_DONE.test(file.getName()))))
+                  .map(RevioPacBioProcessor::getFileCreationTime)
+                  .max(Comparator.naturalOrder());
           // Set completion time based on most recently created .transferdone file
-          dto.setCompletionDate(getFileCreationTime(mostRecentDoneFile.orElse(null)));
+          dto.setCompletionDate(mostRecentDoneFile.get());
```

Comment:
Should probably rename this variable to a time-related name since it's no longer referencing a file